test/client/query-subscribe.js

```diff
@@ -11,596 +12,828 @@ module.exports = function(options) {
+      var query = connection.createSubscribeQuery(
+        'dogs',
+        dbQuery,
+        null,
+        function(err, results) {
           if (err) return done(err);
-          var sorted = util.sortById(results);
-          expect(util.pluck(sorted, 'id')).eql(['fido']);
-          expect(util.pluck(sorted, 'data')).eql([{age: 3}]);
-          connection.get('dogs', 'spot').submitOp({p: ['age'], na: -2});
-        });
-        query.on('error', done);
-        query.on('insert', function(docs, index) {
-          expect(util.pluck(docs, 'id')).eql(['spot']);
-          expect(util.pluck(docs, 'data')).eql([{age: 3}]);
-          expect(index).a('number');
-          var results = util.sortById(query.results);
           expect(util.pluck(results, 'id')).eql(['fido', 'spot']);
-          expect(util.pluck(results, 'data')).eql([{age: 3}, {age: 3}]);
-          done();
+          expect(util.pluck(results, 'data')).eql([{age: 3}, {age: 5}]);
+          connection.get('dogs', 'spot').submitOp({p: ['age'], na: -3});
         });
+      query.on('error', done);
+      query.on('move', function(docs, from, to) {
+        expect(docs.length).eql(1);
+        expect(from).a('number');
+        expect(to).a('number');
+        expect(util.pluck(query.results, 'id')).eql(['spot', 'fido']);
+        expect(util.pluck(query.results, 'data')).eql([{age: 2}, {age: 3}]);
+        done();
       });
     });
+  });
 
-    it('changing a sorted property moves in a subscribed query', function(done) {
-      var connection = this.backend.connect();
-
-      async.parallel([
-        function(cb) {
-          connection.get('dogs', 'fido').on('error', done).create({age: 3}, cb);
-        },
-        function(cb) {
-          connection.get('dogs', 'spot').on('error', done).create({age: 5}, cb);
-        }
-      ], function(err) {
-        if (err) return done(err);
-        var dbQuery = getQuery({query: {}, sort: [['age', 1]]});
-        var query = connection.createSubscribeQuery(
-          'dogs',
-          dbQuery,
-          null,
-          function(err, results) {
-            if (err) return done(err);
-            expect(util.pluck(results, 'id')).eql(['fido', 'spot']);
-            expect(util.pluck(results, 'data')).eql([{age: 3}, {age: 5}]);
-            connection.get('dogs', 'spot').submitOp({p: ['age'], na: -3});
-          });
-        query.on('error', done);
-        query.on('move', function(docs, from, to) {
-          expect(docs.length).eql(1);
-          expect(from).a('number');
-          expect(to).a('number');
-          expect(util.pluck(query.results, 'id')).eql(['spot', 'fido']);
-          expect(util.pluck(query.results, 'data')).eql([{age: 2}, {age: 3}]);
-          done();
-        });
-      });
+  it('returns pubSub error if fails to subscribe to channel', function(done) {
+    sinon.stub(this.backend.pubsub, 'subscribe').callsFake(function(_channel, callback) {
```

Comment:
Let's add an `afterEach` with `sinon.restore` in test/setup.js just to be clean about the tear-down, though Alec points out we create a new backend each test case so it's fine in this case.

Reply:
Added